### PR TITLE
refactor(sc-cli): make cli more generic for non-classic substrate cli

### PR DESCRIPTION
**Pull request by yjhmelody**
- For some cli usage, User do not care about runtime/chain spec infos, so I define  another trait to contain basic cli methods.
Also, for some cli usage, user maybe just want to run a non-classic substrate cli by runner that have their customed config.
- Improve some code details and export some useful functions.

TODO: need to update cumulus/polkadot related code.

--- 
polkadot address: 15ouFh2SHpGbHtDPsJ6cXQfes9Cx1gEFnJJsJVqPGzBSTudr

### CODE REVIEW

File: client/cli/src/lib.rs [hunk 1]
```diff
@@ -183,12 +183,37 @@ pub trait SubstrateCli: Sized {
 		format!("{}/v{}", Self::impl_name(), Self::impl_version())
 	}
 
+	/// Create a generic runner for the config provided in argument. This will pass the config and
+	/// a tokio runtime to runner.
+	fn create_cli_runner<Cfg>(&self, config: Cfg) -> Result<Runner<Cfg>> {
```
Comment (yjhmelody) [inline client/cli/src/lib.rs:188]: Maybe these `create` functions should be renamed.

**Comment by yjhmelody** [comment]
Is it reasonable to change `fn shared_params(&self) -> &SharedParams;` to `fn shared_params(&self) -> Option<&SharedParams>;` ?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2599758

**Comment by yjhmelody** [comment]
Hi, @bkchr Please take a look at how this PR is and see if there is still room for improvement. Ty!

**Comment by yjhmelody** [comment]
@bkchr Anyone give me a feedback?

**Pull request closed without merging**